nexenta-fusion-installer.sh

```diff
@@ -0,0 +1,326 @@
+defaultFusionPath="$HOME/fusion"
+containerName="nexenta-fusion"
+# text formatting variables
+textBold=$(tput bold)
+textNormal=$(tput sgr0)
+textRed='\033[0;31m'
+textBlue='\033[0;36m'
+textYellow='\033[38;5;226m'
+textLightGray='\033[1;37m'
+textNc='\033[0m' # N
+
+ask() {
+    echo
+    # echo question
+    echo -e "${textBlue}$1${textNc}"
+    # echo description
+    echo "----------------"
+    echo -e "${textLightGray}$2${textNc}"
+    echo "----------------"
+}
+
+getTimezone() {
+    if [ -f /etc/timezone ]; then
+        OLSONTZ=$(cat /etc/timezone)
+    elif [ -x "$(sudo command -v systemsetup)" ]; then
+        OLSONTZ=$(sudo systemsetup -gettimezone | grep -Eo '(\w+\/\w+)')
+    elif [ -h /etc/localtime ]; then
+        OLSONTZ=$(readlink /etc/localtime | sed "s/\/usr\/share\/zoneinfo\///")
+    else
+        checksum=$(md5sum /etc/localtime | cut -d' ' -f1)
+        OLSONTZ=$(find /usr/share/zoneinfo/ -type f -exec md5sum {} \; | grep "^$checksum" | sed "s/.*\/usr\/share\/zoneinfo\///" | head -n 1)
+    fi
+}
+
+calculateRAM() {
+    isLowMemory=false
+    if [ -x "/usr/sbin/system_profiler" ]; then
+        # mac os way to find total RAM
+        totalMemory=$(system_profiler SPHardwareDataType | grep "  Memory:" | grep -Eo '[0-9]+')
+        # for mac users we expect that they all have RAM more than 1g and which can be devided by 2
+        # in other cases we set 1g as default
+        if [[ $totalMemory -lt 2 ]]; then
+            isLowMemory=true
+        fi
+        defaultHeapSize=$(( ${totalMemory} / 2 ))g
+        if ! [[ $defaultHeapSize =~ ^[0-9]+g$ ]]; then
+            defaultHeapSize="1g"
+        fi
+        totalMemory=${totalMemory}g
+    elif [ -f "/proc/meminfo" ]; then
+        # linux way to find total RAM
+        totalMemory=$(cat /proc/meminfo | grep -Po "(?<=MemTotal:)(\s+)(\w+)" | grep -Eo "\w+")
+        # converting to bytes
+        # if total memory is lower than 2g
+        if [[ $totalMemory -lt 1024*1024*2 ]]; then
+            isLowMemory=true
+        fi
+        # total memory is presented in kib
+        totalMemory=$(( totalMemory * 1024 ))
+        defaultHeapSize=$(( totalMemory / 2))
+        # 1024^2 = 1048576 // 1mb
+        defaultHeapSize=$( numfmt --to-unit=1048576 --suffix=m ${defaultHeapSize})
+        # converting to human readable value
+        totalMemory=$( numfmt --to=iec ${totalMemory})
+    fi
+}
+
+displayOptions() {
+    local name=$1[@]
+    local options=("${!name}");
+    
+    for ((i=0; i < ${#options[@]}; i++)) {
+        echo "$i) ${options[$i]}"
+    }
+
+    echo "Type a number and press enter"
+}
+
+readSelectedOption() {
+    local name=$1[@]
+    local options=("${!name}");
+
+    local correct=0
+    while [ $correct -eq 0 ]; do 
+        read input
+        if [[ "$input" =~ [0-${#options[@]}] ]]; then
+        #if [ -n "$input" ] && [ $input -ge 0 ] && [ $input -lt ${#options[@]} ]; then
```

Comment:
Is commented code left in a file for a reason?

Reply:
I'll remove it, thanks